src/plugins/auto/auto_schedule.cpp

```diff
@@ -113,18 +113,28 @@ void AutoSchedule::init(const ScheduleContext::Ptr& sContext) {
     }
     _autoSContext->_config[IE::MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES] = _autoSContext->_strDevices;
     std::string profilingTask = "AutoSchedule::AutoSchedule:AutoMode";
-    bool isCumulative = (_autoSContext->_performanceHint == IE::PluginConfigParams::CUMULATIVE_THROUGHPUT) ? true : false;
     // loadContext[ACTUALDEVICE] is always enabled,
     // when there is CPU and there are more than two devices, loadContext[CPU] is enabled
     _loadContext[ACTUALDEVICE].isEnabled = true;
     if (_autoSContext->_modelPath.empty())
         _loadContext[ACTUALDEVICE].networkPrecision = GetNetworkPrecision(_autoSContext->_network);
     _loadContext[ACTUALDEVICE].metaDevices = _autoSContext->_devicePriorities;
+    std::list<DeviceInformation> validDevices;
+    bool isCumuSingleDevice = false;
+    if (_autoSContext->_performanceHint == IE::PluginConfigParams::CUMULATIVE_THROUGHPUT) {
+        validDevices = _autoSContext->_plugin->GetValidDevice(_autoSContext->_devicePriorities,
+                                                              _loadContext[ACTUALDEVICE].networkPrecision);
```

Comment:
if we merge two `if`s:
```suggestion
    bool isCumuSingleDevice = false;
    if (_autoSContext->_performanceHint == IE::PluginConfigParams::CUMULATIVE_THROUGHPUT) {
        auto validDevices = _autoSContext->_plugin->GetValidDevice(_autoSContext->_devicePriorities,
                                                              _loadContext[ACTUALDEVICE].networkPrecision);
```

Reply:
Hi, I have modified the code according to your comments.